setPL.sh

```diff
@@ -205,20 +211,26 @@ verifyTurbostatInstallation
 echo "**** Current PL values from 'turbostat'"
 printTurboStat_PL1_PL2
 
-# set new PL1/PL2 values
-echo "**** Setting PL1=$PL1 and PL2=$PL2 in /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_*_power_limit_uw"
-echo "$PL1" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_0_power_limit_uw
-echo "$PL2" > /sys/class/powercap/intel-rapl/intel-rapl:0/constraint_1_power_limit_uw
+if [ $readOnly -ne $TRUE ]; then
```

Comment:
Rather than having the showOnly checks distributed throughout the logic, please add a block here to specifically handle the read-only case by printing the value of the MMIO (lock status + PL1/PL2 values), then exit. Everything beyond this point will then be only for the set case. You can place the logic in the function to make it reusable for both the showOnly and regular cases.